### PR TITLE
Remove .impl_UNBOXED() and functionalities associated with it

### PR DESCRIPTION
**Pull request by smessmer**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #49613 Remove backward and requires_grad from Autograd backend key
* #49259 Remove `use_c10_dispatcher: full` lines
* #49251 Remove generated_unboxing_wrappers and setManuallyBoxedKernel
* **#49220 Remove .impl_UNBOXED() and functionalities associated with it**
* #49164 Remove codegen logic to support non-c10-full ops

Since all ops are c10-full, we can remove .impl_UNBOXED now.
This also removes the ability of KernelFunction or CppFunction to store unboxedOnly kernels.

Differential Revision: [D25490225](https://our.internmc.facebook.com/intern/diff/D25490225/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25490225/)!

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
Good riddance!!!

**Comment by ezyang** [comment]
Guessing this will need an XLA companion?

**Comment by smessmer** [comment]
> Guessing this will need an XLA companion?

yes, https://github.com/pytorch/xla/pull/2714 needs to be landed before this

**Review by bhosmer — APPROVED** [review]

File: torch/library.h [hunk 1]
```diff
@@ -543,7 +537,7 @@ class TORCH_API Library final {
   }
   template <typename Func>
   Library& impl_UNBOXED(detail::SelectiveStr<true> name, Func* raw_f) & {
-    return impl(name.operator const char*(), CppFunction::makeUnboxedOnly(raw_f));
+    static_assert(c10::guts::false_t<Func>(), ".impl_UNBOXED(...) was removed. Please use .impl(...) instead.");
```
Comment (bhosmer) [inline torch/library.h:540]: FYI looks like the lack of return statement here is the cause of some/all of your build errors

**Pull request closed without merging**